### PR TITLE
[Documentation] Fixing Typescript compiler option and cleaning up the…

### PR DESCRIPTION
**Pull request by rajkaramchedu**
This PR makes the below changes:

1. Provides an allowed value for the `module` option in the Typescript version. I got an error and it did not run with the default value. 
2. Adds a "before you start" section.

### CODE REVIEW

File: developer-docs-site/docs/tutorials/first-transaction.md [hunk 1]
```diff
@@ -1,50 +1,90 @@
...
-This tutorial outlines how to generate, submit, and verify transactions submitted to the Aptos Blockchain. The steps for doing so are:
+This tutorial describes, in the following step-by-step approach, how to generate, submit, and verify transactions submitted to the Aptos Blockchain:
 
-1. Create a representation of an account
-2. Prepare a wrapper around the REST interfaces
-3. Prepare a wrapper around the Faucet interface
-4. Combine them into an application, execute and verify
+1. Create a representation of an account.
+  
+  Each Aptos account has a unique account address. The owner of that account holds the public, private key-pair that maps to the Aptos account address and, in turn, the authentication key stored in that account.
```
Comment (movekevin) [inline developer-docs-site/docs/tutorials/first-transaction.md:16]: Should we also mention that the authentication key can be changed without changing the account address?

File: developer-docs-site/docs/tutorials/first-transaction.md [hunk 2]
```diff
@@ -1,50 +1,90 @@
...
+
+  Using the Faucet interface at the Aptos devnet, this tutorial code automatically creates an account with the account address `0x1` and funds the account. 
+
+4. Combine the above wrappers into an application, execute and verify.
+
+## Before you start
+
+Make sure you follow the below steps first so you can run the tutorial. 
+
+1. Clone the Aptos repo.
```
Comment (movekevin) [inline developer-docs-site/docs/tutorials/first-transaction.md:36]: We should also mention that people should be checking out the devnet branch here. There might be some incompatibility problems if they don't do so.

File: developer-docs-site/docs/tutorials/first-transaction.md [hunk 3]
```diff
@@ -70,9 +110,9 @@ Each Aptos account has a unique account address. The owner of that account holds
   </TabItem>
 </Tabs>
 
-## Step 2) REST interface
+## Step 2: REST interface
 
-Aptos exposes a [REST interface][rest_spec] for interacting with the blockchain. While the data from the REST interface can be read directly, the following snippets of code demonstrate a more ergonomic approach. This next set of code snippets demonstrates how to use the REST interface to retrieve ledger data from the FullNode including account and account resource data. It also demonstrates how to use the REST interface for constructing signed transactions represented by JSON formatting.
+ While the data from the REST interface can be read directly, the following code examples demonstrate a more ergonomic approach. The below code examples demonstrate how to use the REST interface to retrieve ledger data from the FullNode including account and account resource data. The code also demonstrates how to use the REST interface for constructing signed transactions represented by JSON formatting.
```
Comment (movekevin) [inline developer-docs-site/docs/tutorials/first-transaction.md:115]: nit: A bit redundant: "the following code examples demonstrate a more ergonomic approach. The below code examples demonstrate"

File: developer-docs-site/docs/tutorials/first-transaction.md [hunk 4]
```diff
@@ -181,9 +222,9 @@ The following demonstrate how to read data from the blockchain and how to write
   </TabItem>
 </Tabs>
 
-## Step 3) Faucet interface
+## Step 3: Faucet interface
 
-Blockchain faucets issue test tokens to accounts. These can be used for testing, e.g., paying gas fees or transferring tokens between users. The Aptos faucet can also create accounts if they do not exist yet. The Aptos faucet interface requires a public key represented in a hex-encoded string.
+Aptos Blockchain faucets issue test tokens to accounts. These test tokens can be used for testing, e.g., paying gas fees or transferring tokens between users. The Aptos Faucet can also create accounts if they do not exist. The Aptos Faucet interface requires a public key represented in a hex-encoded string.
```
Comment (movekevin) [inline developer-docs-site/docs/tutorials/first-transaction.md:227]: What do you think about the terminologies here: coins vs tokens? I think we've been using them differently (coins = tokens, tokens = NFTs) from the rest of the crypto world (coin = native coin on a network, token = ERC-20 style interfaces)

File: developer-docs-site/static/examples/typescript/tsconfig.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
     "types": [
       "node"
     ],
-    "module": "node12",
```
Comment (movekevin) [inline developer-docs-site/static/examples/typescript/tsconfig.json:10]: Why this change?

**Review by movekevin — COMMENTED** [review]
Thanks for cleaning this up!

File: developer-docs-site/static/examples/typescript/tsconfig.json (hunk 5 above)
Comment (rajkaramchedu) [reply developer-docs-site/static/examples/typescript/tsconfig.json:10]: When I ran it on my mac book m1, it errored out saying `TSError: ⨯ Unable to compile TypeScript:
error TS6046: Argument for '--module' option must be: 'none', 'commonjs', 'amd', 'system', 'umd', 'es6', 'es2015', 'es2020', 'es2022', 'esnext', 'node16', 'nodenext'` and I didn't understand what the correct value should be. Looking up here https://www.typescriptlang.org/tsconfig#module saw the suggestion and it worked. I don't know Typescript enough to say what should be the value there.

File: developer-docs-site/docs/tutorials/first-transaction.md (hunk 1 above)
Comment (rajkaramchedu) [reply developer-docs-site/docs/tutorials/first-transaction.md:16]: I think probably we should give that detail in the Accounts document. Here in this tutorial, strictly speaking, the user is not even explicitly creating an account and it might be jumping the gun if we mention changing auth keys. 
I am going to take another thorough look at the Accounts doc also separately.

File: developer-docs-site/docs/tutorials/first-transaction.md (hunk 2 above)
Comment (rajkaramchedu) [reply developer-docs-site/docs/tutorials/first-transaction.md:36]: Good point.

File: developer-docs-site/docs/tutorials/first-transaction.md (hunk 4 above)
Comment (rajkaramchedu) [reply developer-docs-site/docs/tutorials/first-transaction.md:227]: I have it in my to-do list to address this. Totally agree. I want to address it at an overall higher level and do the change in all our docs.

File: developer-docs-site/static/examples/typescript/tsconfig.json (hunk 5 above)
Comment (movekevin) [reply developer-docs-site/static/examples/typescript/tsconfig.json:10]: Did you run yarn? If yes, may be make sure you have the right nodejs version installed. I just tried running this at HEAD on my laptop (also an M1) and it works fine.

Comment (movekevin) [reply developer-docs-site/static/examples/typescript/tsconfig.json:10]: I tried compiling with commonjs and it works so may be it's fine to keep it updated.
